Remove experimental note of tiflash disaggregated arch (#14859)
pingcap/docs: tiflash/tiflash-disaggregated-and-s3.md

    You can also use other S3-compatible object storage, such as [MinIO](https://min.io/).

>     Make sure the access permissions of S3 APIs are granted. The S3 APIs used by TiFlash are as follows:

Author: ```suggestion
    TiFlash will use the following S3 APIs for accessing data. Make sure that the nodes deployed with TiFlash have the necessary permissions for these APIs.
```

    - GetObjectTagging
    - PutBucketLifecycle

2. Add a [lifecycle](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html) to the prepared S3 bucket for cleaning up deleted data:

    ```shell
    "ID": "tiflash-clean",
    "Expiration": {
        "Days": 1
    },
    "Filter": {
        "And": {
            "Tags": [
                {
                    "Value": "tiflash_deleted", 
                    "Key": "true"
                }
            ]
        }
    }

>     ```

Author: Remove because tiflash can automatically set it

> 2. Make sure that there are no TiFlash nodes is deployed using the coupled storage and compute architecture in the TiDB cluster. If any, set the TiFlash replica count of all tables to `0` and then remove all TiFlash nodes. For example:

Reviewer: ```suggestion
2. Make sure your TiDB cluster has no TiFlash nodes deployed using the coupled storage and compute architecture. If any, set the TiFlash replica count of all tables to `0` and then remove all TiFlash nodes. For example:
```

## Restrictions

> - TiFlash does not support in-place switching between the **disaggregated storage and compute architecture** and the **coupled storage and compute architecture**. Before switching architectures, you must remove all existing coupled architecture TiFlash nodes.

Reviewer: ```suggestion
- TiFlash does not support in-place switching between the **disaggregated storage and compute architecture** and the **coupled storage and compute architecture**. Before switching to the disaggregated architecture, you must remove all existing TiFlash nodes deployed using the coupled architecture.
```
